src/BonjourMDNSManager.cpp

```diff
@@ -464,6 +466,10 @@ class MDNSManager::PImpl
                 removeTrailingDot(rr->domain);
                 removeTrailingDot(host);
 
+                if (host_struct->h_length > 0) {
+                    std::string hostip = inet_ntoa(**(struct in_addr **) host_struct->h_addr_list);
```

Comment:
Is blindly picking the first address safe?

I'm not sure I trust the thread safety of `inet_ntoa` (uses an internal static char buffer). Consider using `inet_ntop` instead.